### PR TITLE
[CELEBORN-1005][BUG] Clean Expired App Dirs will delete the running a…

### PR DESCRIPTION
**Pull request by wilsonjie**
### What changes were proposed in this pull request?
When working on reading shuffle data, the file was accidentally deleted

`2023-09-22 16:32:36,810 [storage-scheduler] INFO  org.apache.celeborn.service.deploy.worker.storage.StorageManager[51]: Delete expired app dir /data8/rssdata/celeborn-worker/shuffle_data/application_1689848866482_12296544_1.
2023-09-22 16:32:36,810 [Disk-cleaner-/data8-6] DEBUG org.apache.celeborn.service.deploy.worker.storage.StorageManager[47]: Deleted expired shuffle file /data8/rssdata/celeborn-worker/shuffle_data/application_1689848866482_12296544_1/32/924-0-0.
2023-09-22 16:32:53,304 [fetch-server-11-31] DEBUG org.apache.celeborn.service.deploy.worker.FetchHandler[47]: Received chunk fetch request application_1689848866482_12296544_1-32 924-0-0 0 2147483647 get file info FileInfo{file=/data8/rssdata/celeborn-worker/shuffle_data/application_1689848866482_12296544_1/32/924-0-0, chunkOffsets=0,558, userIdentifier=`default`.`default`, partitionType=REDUCE}
java.io.FileNotFoundException: /data8/rssdata/celeborn-worker/shuffle_data/application_1689848866482_12296544_1/32/924-0-0 (No such file or directory)`

Because when cleaning up the directories of expired apps, the file directory is created first and then added to the fileInfos collection. As a result, when getting the shuffleKeySet, the running apps do not yet exist, causing the files to be mistakenly deleted.

https://issues.apache.org/jira/browse/CELEBORN-1005
### Why are the changes needed?
bugfix

### How was this patch tested?

### CODE REVIEW

File: worker/src/main/scala/org/apache/celeborn/service/deploy/worker/storage/StorageManager.scala [hunk 1]
```diff
@@ -358,13 +358,14 @@ final private[worker] class StorageManager(conf: CelebornConf, workerSource: Abs
...
         val shuffleDir =
           new Path(new Path(hdfsDir, conf.workerWorkingDir), s"$appId/$shuffleId")
-        FileSystem.mkdirs(StorageManager.hadoopFs, shuffleDir, hdfsPermission)
         val fileInfo =
           new FileInfo(
             new Path(shuffleDir, fileName).toString,
             userIdentifier,
             partitionType,
             partitionSplitEnabled)
+        fileInfos.computeIfAbsent(shuffleKey, newMapFunc).put(fileName, fileInfo)
```
Comment (waitinfuture) [inline worker/src/main/scala/org/apache/celeborn/service/deploy/worker/storage/StorageManager.scala:367]: Before this PR, `fileInfos` will not put a new object if Exception is thrown. So I think we'd better try to delete the `fileName` from `fileInfos` where `createWriter` is called, i.e. in the catch block of `Controller.handleReserveSlots`

File: worker/src/main/scala/org/apache/celeborn/service/deploy/worker/storage/StorageManager.scala [hunk 2]
```diff
@@ -399,6 +399,14 @@ final private[worker] class StorageManager(conf: CelebornConf, workerSource: Abs
...
         val file = new File(shuffleDir, fileName)
         try {
+          val fileInfo =
+            new FileInfo(
+              file.getAbsolutePath,
+              userIdentifier,
+              partitionType,
+              partitionSplitEnabled)
+          fileInfo.setMountPoint(mountPoint)
+          fileInfos.computeIfAbsent(shuffleKey, newMapFunc).put(fileName, fileInfo)
```
[the remark this answers is not in the document]
Comment (wilsonjie) [reply worker/src/main/scala/org/apache/celeborn/service/deploy/worker/storage/StorageManager.scala:409]: fixed

File: worker/src/main/scala/org/apache/celeborn/service/deploy/worker/storage/StorageManager.scala (hunk 1 above)
Comment (wilsonjie) [reply worker/src/main/scala/org/apache/celeborn/service/deploy/worker/storage/StorageManager.scala:367]: During the heartbeat between master and worker, the active ShuffleKeys of the worker are compared with the registered ShuffleKeys of the master, and expired ones are added to the expired ShuffleKeys for automatic cleaning.

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks! Merging to main/0.3

**Pull request closed without merging**